Automattic/newspack-blocks
# feat(donate-block): hide buttons if only one frequency is available

### All Submissions:






### Changes proposed in this Pull Request:



If the Donate block is set to render just one frequency, the frequency buttons section should not be displayed.
Instead, a frequency label is added, if applicable.

| Before | After | 
|--------------|-----------|
| <img width="810" alt="image" src="https://github.com/user-attachments/assets/d20f1727-ee7c-4fc3-8b2b-93a799db5961"> | <img width="812" alt="image" src="https://github.com/user-attachments/assets/b4798c3c-2c20-4052-87bb-b7d9fc1bc368">
| <img width="814" alt="image" src="https://github.com/user-attachments/assets/221e49fe-933e-4f9c-86bb-78f834848eac"> | <img width="804" alt="image" src="https://github.com/user-attachments/assets/c3927b64-7919-4345-a135-b57ee9e6d231">

### How to test the changes in this Pull Request:

1. On `trunk`, insert Donate blocks (see below for a copy-paste)
3. Observe the frequency choosing UI is rendered, with a single button
4. Switch to this branch, reload editor & frontend – observe the selection UI is not rendered and frequency labels added instead, where applicable

<details>
<summary>
Blocks code for testing
</summary>
<pre>
&lt;!-- wp:newspack-blocks/donate {"className":"is-style-default","manual":true,"amounts":{"once":[9,20,90,20],"month":[7,15,30,15],"year":[84,180,360,180]},"disabledFrequencies":{"once":false,"month":true,"year":true},"defaultFrequency":"once","minimumDonation":5} /--&gt;

&lt;!-- wp:newspack-blocks/donate {"className":"is-style-default","manual":true,"amounts":{"once":[9,20,90,20],"month":[7,15,30,15],"year":[84,180,360,180]},"disabledFrequencies":{"once":true,"month":false,"year":true},"minimumDonation":5} /--&gt;

&lt;!-- wp:newspack-blocks/donate {"className":"is-style-default","manual":true,"amounts":{"once":[9,20,90,20],"month":[7,15,30,15],"year":[84,180,360,180]},"disabledFrequencies":{"once":true,"month":true,"year":false},"defaultFrequency":"year","minimumDonation":5} /--&gt;

&lt;!-- wp:newspack-blocks/donate {"className":"is-style-default","manual":true,"tiered":false,"amounts":{"once":[9,20,90,20],"month":[7,15,30,15],"year":[84,180,360,180]},"disabledFrequencies":{"once":false,"month":true,"year":true},"defaultFrequency":"once","minimumDonation":5} /--&gt;

&lt;!-- wp:newspack-blocks/donate {"className":"is-style-default","manual":true,"tiered":false,"amounts":{"once":[9,20,90,20],"month":[7,15,30,15],"year":[84,180,360,180]},"disabledFrequencies":{"once":true,"month":false,"year":true},"minimumDonation":5} /--&gt;

&lt;!-- wp:newspack-blocks/donate {"className":"is-style-default","manual":true,"tiered":false,"amounts":{"once":[9,20,90,20],"month":[7,15,30,15],"year":[84,180,360,180]},"disabledFrequencies":{"once":true,"month":true,"year":false},"defaultFrequency":"year","minimumDonation":5} /--&gt;
</pre>
</details>

### Other information:




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208106329606187
